modular_skyrat/master_files/code/modules/client/preferences/mutant_parts.dm

```diff
@@ -0,0 +1,1292 @@
+
+/// IPC Chassis
+
+/datum/preference/toggle/ipc_chassis
+	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "ipc_chassis_toggle"
+	relevant_mutant_bodypart = "ipc_chassis"
+	default_value = FALSE
+
+/datum/preference/toggle/ipc_chassis/apply_to_human(mob/living/carbon/human/target, value, datum/preferences/preferecnes)
+	return TRUE // we dont actually want this to do anything
+
+/datum/preference/toggle/ipc_chassis/is_accessible(datum/preferences/preferences)
+	var/passed_initial_check = ..(preferences)
+	var/allowed = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
+	return passed_initial_check || allowed
+
+/datum/preference/choiced/ipc_chassis
+	savefile_key = "feature_ipc_chassis"
+	savefile_identifier = PREFERENCE_CHARACTER
+	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+	relevant_mutant_bodypart = "ipc_chassis"
+
+/datum/preference/choiced/ipc_chassis/is_accessible(datum/preferences/preferences)
+	var/passed_initial_check = ..(preferences)
+	var/allowed = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
+	var/part_enabled = preferences.read_preference(/datum/preference/toggle/ipc_chassis)
+	return ((passed_initial_check || allowed) && part_enabled)
+
+/datum/preference/choiced/ipc_chassis/init_possible_values()
+	return assoc_to_keys(GLOB.sprite_accessories["ipc_chassis"])
+
+/datum/preference/choiced/ipc_chassis/apply_to_human(mob/living/carbon/human/target, value)
+	if(!target.dna.mutant_bodyparts["ipc_chassis"])
+		target.dna.mutant_bodyparts["ipc_chassis"] = list("name" = "None", "color" = list("FFF", "FFF", "FFF"))
+	target.dna.mutant_bodyparts["ipc_chassis"]["name"] = value
+
+/datum/preference/choiced/ipc_chassis/create_default_value()
+	var/datum/sprite_accessory/ipc_chassis/default = /datum/sprite_accessory/ipc_chassis/none
+	return initial(default.name)
+
+/datum/preference/tri_color/ipc_chassis
+	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "ipc_chassis_color"
+	relevant_mutant_bodypart = "ipc_chassis"
+
+/datum/preference/tri_color/ipc_chassis/is_accessible(datum/preferences/preferences)
+	var/passed_initial_check = ..(preferences)
+	var/allowed = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
+	var/part_enabled = preferences.read_preference(/datum/preference/toggle/ipc_chassis)
+	return ((passed_initial_check || allowed) && part_enabled)
+
+/datum/preference/tri_color/ipc_chassis/apply_to_human(mob/living/carbon/human/target, value)
+	if(!target.dna.mutant_bodyparts["ipc_chassis"])
+		target.dna.mutant_bodyparts["ipc_chassis"] = list("name" = "None", "color" = list("FFF", "FFF", "FFF"))
+	target.dna.mutant_bodyparts["ipc_chassis"]["color"] = list(sanitize_hexcolor(value[1]), sanitize_hexcolor(value[2]), sanitize_hexcolor(value[3]))
+
+/// Neck Accessories
+
+/datum/preference/toggle/neck_acc
```

Comment:
Same as for head accessories, you should write the full thing, shortening it doesn't really add much and reduces, even if only slightly, readability.